src/NServiceBus.Core.Tests/TransportDispatch/HeaderAssertHelpers.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace NServiceBus.Core.Tests
+{
+    using System;
+    using NServiceBus.Pipeline.Contexts;
+
+    static class HeaderAssertHelpers
+    {
+        public static void AssertHeaderWasSet(this PhysicalOutgoingContextStageBehavior.Context context, string key, Predicate<string> predicate)
+        {
+            Assert(context.Extensions.GetOrCreate<DispatchMessageToTransportConnector.State>(), key, predicate);
+        }
+
+        public static void AssertHeaderWasSet(this OutgoingContext context, string key, Predicate<string> predicate)
+        {
+            Assert(context.Extensions.GetOrCreate<DispatchMessageToTransportConnector.State>(), key, predicate);
+        }
+
+        static void Assert(DispatchMessageToTransportConnector.State state, string key, Predicate<string> predicate)
```

Comment:
No preference, jon skeet seems to imply predicate? http://stackoverflow.com/questions/566860/delegates-predicate-action-func